src/ansys/geometry/core/geometry/box_uv.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Provides the ``BoxUV`` class."""
+from enum import Enum
+
+from ansys.geometry.core.geometry.parameterization import Interval, ParamUV
+from ansys.geometry.core.typing import Real
+
+
+class LocationUV(Enum):
+    """Provides the ``LocationUV`` class to indicate locations for BoxUV."""
+
+    TopLeft = 1
+    TopCenter = 2
+    TopRight = 3
+    BottomLeft = 4
+    BottomCenter = 5
+    BottomRight = 6
+    LeftCenter = 7
+    RightCenter = 8
+    Center = 9
+
+
+class BoxUV:
+    """BoxUV class."""
+
+    def __init__(self, range_u: Interval = None, range_v: Interval = None) -> None:
+        """Root constructor for BoxUV."""
+        if range_u is not None:
+            self._interval_u = range_u
+        if range_v is not None:
+            self._interval_v = range_v
+
+    @classmethod
+    def from_param(cls, param: ParamUV):
+        """Secondary constructor for BoxUV using a ParamUV object type."""
```

Comment:
Are this entry and the next entry methods? Methods should be written as a verb with no "s" or "es"--maybe something like "Create a secondary constructor for a X using a ``ParamUV`` object type.

The X should use plain English.